docs/reference/type-system.md

```diff
@@ -0,0 +1,27 @@
+# Type System
+
+## Motivation
+
+Feast uses an internal type system to provide guarantees on training and serving data.
+Feast currently supports eight primitive types - `INT32`, `INT64`, `FLOAT32`, `FLOAT64`, `STRING`, `BYTES`, `BOOL`, and `UNIX_TIMESTAMP` - and the corresponding array types.
```

Comment:
What about `NULL`? And are any of these types nullable, or must they be pure values?